datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -304,154 +304,75 @@ fn summarize_min_max(
-                            vec![Some(*s.max())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                max_values[i] = None;
-                            }
-                        }
-                    }
-                    if let Some(min_value) = &mut min_values[i] {
-                        match min_value.update_batch(&[Arc::new(Float32Array::from(
-                            vec![Some(*s.min())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                min_values[i] = None;
-                            }
-                        }
-                    }
-                    return;
-                }
+        ParquetStatistics::Float(s)
+            if DataType::Float32 == *fields[i].data_type() && s.has_min_max_set() =>
+        {
+            if let Some(max_value) = &mut max_values[i] {
+                max_value
+                    .update_batch(&[Arc::new(Float32Array::from(vec![Some(*s.max())]))])
+                    .unwrap_or_else(|_| max_values[i] = None);
+            }
+            if let Some(min_value) = &mut min_values[i] {
+                min_value
+                    .update_batch(&[Arc::new(Float32Array::from(vec![Some(*s.min())]))])
+                    .unwrap_or_else(|_| min_values[i] = None);
             }
-            max_values[i] = None;
-            min_values[i] = None;
         }
-        ParquetStatistics::Double(s) => {
-            if let DataType::Float64 = fields[i].data_type() {
-                if s.has_min_max_set() {
-                    if let Some(max_value) = &mut max_values[i] {
-                        match max_value.update_batch(&[Arc::new(Float64Array::from(
-                            vec![Some(*s.max())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                max_values[i] = None;
-                            }
-                        }
-                    }
-                    if let Some(min_value) = &mut min_values[i] {
-                        match min_value.update_batch(&[Arc::new(Float64Array::from(
-                            vec![Some(*s.min())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                min_values[i] = None;
-                            }
-                        }
-                    }
-                    return;
-                }
+        ParquetStatistics::Double(s)
+            if DataType::Float64 == *fields[i].data_type() && s.has_min_max_set() =>
+        {
+            if let Some(max_value) = &mut max_values[i] {
+                max_value
+                    .update_batch(&[Arc::new(Float64Array::from(vec![Some(*s.max())]))])
```

Comment:
```suggestion
                    .update_batch(&[Arc::new(Float64Array::from(vec![*s.max()]))])
```

I think this also works? Goes for the other branches too